### DISCUSSION TITLE
Can't load the application after version update (1.2.2 to 1.8 preview)

### DISCUSSION THREAD

**Discussion by khurram4u2** [OrchardCMS/OrchardCore#14931] [General]
### Describe the bug
Hi,

I have updated my application from orchard core 1.2 to 1.8 preview and getting following errors. Could you please help?

`17:19:44 INF] Executed action OrchardCore.Contents.Controllers.ItemController.Display (OrchardCore.Contents) in 1733.97ms
[17:19:44 INF] Executed endpoint 'OrchardCore.Contents.Controllers.ItemController.Display (OrchardCore.Contents)'
[17:19:44 ERR] An unhandled exception has occurred while executing the request.
Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.CompilationFailedException: One or more compilation failures occurred:yqnaboey.vlg(4,20): error CS0400: The type or namespace name 'Microsoft' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(4,82): error CS0518: Predefined type 'System.Type' is not defined or imported
yqnaboey.vlg(4,149): error CS0518: Predefined type 'System.String' is not defined or imported
yqnaboey.vlg(4,166): error CS0518: Predefined type 'System.String' is not defined or imported
yqnaboey.vlg(8,11): error CS0246: The type or namespace name 'System' could not be found (are you missing a using directive or an assembly reference?)
yqnaboey.vlg(9,11): error CS0246: The type or namespace name 'System' could not be found (are you missing a using directive or an assembly reference?)
yqnaboey.vlg(10,11): error CS0246: The type or namespace name 'System' could not be found (are you missing a using directive or an assembly reference?)
yqnaboey.vlg(11,11): error CS0246: The type or namespace name 'System' could not be found (are you missing a using directive or an assembly reference?)
yqnaboey.vlg(12,11): error CS0246: The type or namespace name 'Microsoft' could not be found (are you missing a using directive or an assembly reference?)
yqnaboey.vlg(13,11): error CS0246: The type or namespace name 'Microsoft' could not be found (are you missing a using directive or an assembly reference?)
yqnaboey.vlg(14,11): error CS0246: The type or namespace name 'Microsoft' could not be found (are you missing a using directive or an assembly reference?)
yqnaboey.vlg(211,14): error CS0400: The type or namespace name 'Microsoft' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(212,14): error CS0400: The type or namespace name 'Microsoft' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(213,14): error CS0400: The type or namespace name 'Microsoft' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(214,14): error CS0400: The type or namespace name 'System' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(211,78): error CS0518: Predefined type 'System.String' is not defined or imported
yqnaboey.vlg(211,87): error CS0518: Predefined type 'System.String' is not defined or imported
yqnaboey.vlg(211,132): error CS0518: Predefined type 'System.String' is not defined or imported
yqnaboey.vlg(212,78): error CS0518: Predefined type 'System.String' is not defined or imported
yqnaboey.vlg(212,87): error CS0518: Predefined type 'System.String' is not defined or imported
yqnaboey.vlg(212,132): error CS0518: Predefined type 'System.String' is not defined or imported
yqnaboey.vlg(213,84): error CS0518: Predefined type 'System.String' is not defined or imported
yqnaboey.vlg(213,98): error CS0518: Predefined type 'System.String' is not defined or imported
yqnaboey.vlg(215,67): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
yqnaboey.vlg(215,113): error CS1980: Cannot define a class or member that utilizes 'dynamic' because the compiler required type 'System.Runtime.CompilerServices.DynamicAttribute' cannot be found. Are you missing a reference?
yqnaboey.vlg(215,113): error CS0518: Predefined type 'System.Boolean' is not defined or imported
yqnaboey.vlg(218,39): error CS0400: The type or namespace name 'System' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(240,24): error CS0400: The type or namespace name 'Microsoft' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(240,118): error CS0518: Predefined type 'System.Void' is not defined or imported
yqnaboey.vlg(242,24): error CS0400: The type or namespace name 'Microsoft' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(242,71): error CS0518: Predefined type 'System.Void' is not defined or imported
yqnaboey.vlg(244,24): error CS0400: The type or namespace name 'Microsoft' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(244,87): error CS0518: Predefined type 'System.Void' is not defined or imported
yqnaboey.vlg(246,24): error CS0400: The type or namespace name 'Microsoft' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(246,83): error CS0518: Predefined type 'System.Void' is not defined or imported
yqnaboey.vlg(248,24): error CS0400: The type or namespace name 'Microsoft' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(248,71): error CS0518: Predefined type 'System.Object' is not defined or imported
yqnaboey.vlg(248,71): error CS1980: Cannot define a class or member that utilizes 'dynamic' because the compiler required type 'System.Runtime.CompilerServices.DynamicAttribute' cannot be found. Are you missing a reference?
yqnaboey.vlg(248,71): error CS0518: Predefined type 'System.Boolean' is not defined or imported
yqnaboey.vlg(248,92): error CS0518: Predefined type 'System.Void' is not defined or imported
yqnaboey.vlg(218,67): error CS0115: '_Areas_OrchardCore_Contents_Views_Item_Display.ExecuteAsync()': no suitable method found to override
yqnaboey.vlg(239,18): error CS0400: The type or namespace name 'Microsoft' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(241,18): error CS0400: The type or namespace name 'Microsoft' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(243,18): error CS0400: The type or namespace name 'Microsoft' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(245,18): error CS0400: The type or namespace name 'Microsoft' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(247,18): error CS0400: The type or namespace name 'Microsoft' could not be found in the global namespace (are you missing an assembly reference?)
yqnaboey.vlg(236,13): error CS0518: Predefined type 'System.Object' is not defined or imported
yqnaboey.vlg(236,13): error CS0103: The name 'WriteLiteral' does not exist in the current context
yqnaboey.vlg(236,26): error CS0518: Predefined type 'System.String' is not defined or imported
yqnaboey.vlg(218,67): error CS0161: '_Areas_OrchardCore_Contents_Views_Item_Display.ExecuteAsync()': not all code paths return a value
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(5,7): error CS0246: The type or namespace name 'Microsoft' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(6,7): error CS0246: The type or namespace name 'Microsoft' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(7,7): error CS0246: The type or namespace name 'Microsoft' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(8,7): error CS0246: The type or namespace name 'Microsoft' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(9,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(11,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(12,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(13,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(14,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(15,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(16,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(17,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(18,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(19,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(20,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(21,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(22,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(23,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(24,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(25,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(26,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(27,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(28,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(30,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(31,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/_ViewImports.cshtml(32,7): error CS0246: The type or namespace name 'System' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/Item/Display.cshtml(1,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/Item/Display.cshtml(2,7): error CS0246: The type or namespace name 'OrchardCore' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/Item/Display.cshtml(4,5): error CS0518: Predefined type 'System.Object' is not defined or imported
/Areas/OrchardCore.Contents/Views/Item/Display.cshtml(4,5): error CS0246: The type or namespace name 'ContentItem' could not be found (are you missing a using directive or an assembly reference?)
/Areas/OrchardCore.Contents/Views/Item/Display.cshtml(4,31): error CS0518: Predefined type 'System.Object' is not defined or imported
/Areas/OrchardCore.Contents/Views/Item/Display.cshtml(4,31): error CS0103: The name 'Model' does not exist in the current context
/Areas/OrchardCore.Contents/Views/Item/Display.cshtml(7,1): error CS0518: Predefined type 'System.Object' is not defined or imported
/Areas/OrchardCore.Contents/Views/Item/Display.cshtml(7,1): error CS0103: The name 'Write' does not exist in the current context
/Areas/OrchardCore.Contents/Views/Item/Display.cshtml(7,13): error CS0518: Predefined type 'System.Object' is not defined or imported
/Areas/OrchardCore.Contents/Views/Item/Display.cshtml(7,13): error CS0103: The name 'DisplayAsync' does not exist in the current context
/Areas/OrchardCore.Contents/Views/Item/Display.cshtml(7,26): error CS0518: Predefined type 'System.Object' is not defined or imported
/Areas/OrchardCore.Contents/Views/Item/Display.cshtml(7,26): error CS0103: The name 'Model' does not exist in the current context
   at Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.RuntimeViewCompiler.CompileAndEmit(RazorCodeDocument codeDocument, String generatedCode)
   at Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.RuntimeViewCompiler.CompileAndEmit(String relativePath)
   at Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation.RuntimeViewCompiler.OnCacheMiss(String normalizedPath)`

### To Reproduce
1. Go to 'update from version 1.2 to 1.8.0 preview'
2. Click on '...'
3. Scroll down to '...'

### Expected behavior
Application was suppose to load default page.

### Screenshots
.
![image](https://github.com/OrchardCMS/OrchardCore/assets/56805276/02e9617d-bfed-4009-8e07-61f4c4dd9a7c)

**Comment by hishamco** [comment]
I recently updated OC 1.0,0-beta to 1.7.0 and it works, but it depends on what the modules you have. I advice you to do an increamental update instead

**Comment by MikeAlhayek** [comment]
This sounds like a compile time error. Isn't it? The errors are saying your missing using statement.

Make sure your project is using .net 8, since we dropped support to other frameworks in OC 1.8-previews. Then make sure you are able to build the project locally and fix any compile time errors.
